### ISSUE TITLE
Extending parent theme template breaks dependant templates

### ISSUE THREAD

**Issue by rmilecki** [PrestaShop/PrestaShop#30712]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

According to the documentation it should be possible to extend parent theme to introduce small changes (e.g. modify a single template block):
https://devdocs.prestashop-project.org/1.7/themes/reference/template-inheritance/parent-child-feature/

In documentation example `product-list.tpl` gets extended.

The problem (bug) is that extending `product-list.tpl` breaks templates that extend `product-list.tpl` (e.g. `category.tpl`).

### Expected behavior

Extending `product-list.tpl` should allow modifying that template without affecting behaviour of any other templates.

### Steps to reproduce

1. Go to the Design → Theme & Logo
2. Use `Add new theme`
3. Use `Import from your computer` to install [childtest.zip](https://github.com/PrestaShop/PrestaShop/files/10329101/childtest.zip)
4. Switch to the "Child Theme Test" theme

### PrestaShop version(s) where the bug happened

1.7.8.8 8.0.0

### PHP version(s) where the bug happened

7.2.34 7.3.32

**Comment by rmilecki** [comment]
My `childtest` theme contains a very simple `product-list.tpl` file:
```
{extends file='parent:catalog/listing/product-list.tpl'}

{block name='content' prepend}
  <h1>Child Theme test</h1>
{/block}
```

---

## Original behaviour
![original](https://user-images.githubusercontent.com/452972/210178587-26f6702d-92ba-4161-bf07-e579781eabf0.png)

## Expected behaviour
![expected](https://user-images.githubusercontent.com/452972/210178595-16362094-07a3-48f3-a6e3-523e79a283a3.png)

## Actual (bugged) behaviour
![bugged](https://user-images.githubusercontent.com/452972/210178600-ed0fe63a-795e-4db2-a3e0-bb00caf00100.png)

As you can see extending `product-list.tpl` (e.g. to modify an existing block) breaks `themes/classic/templates/catalog/listing/category.tpl`. For some reason `category.tpl` which provides a nice category header doesn't work anymore.

**Comment by AureRita** [comment]
Hi @rmilecki ,

I reproduce the issue with PrestaShop version 1.7.7.8, I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

It's maybe because you didn't have a `category.tpl` on your childtest.zip that you have this issue ( if I add you line on the `product-list.tpl` I can't go on a page with the faceted search as we can see here ) 

![image](https://user-images.githubusercontent.com/100699445/210333082-f13d9ca3-c656-4a8c-a32c-b8575a5dfa0c.png)

Thanks

**Comment by rmilecki** [comment]
> It's maybe because you didn't have a category.tpl on your childtest.zip that you have this issue

That's correct, adding all dependant templates to the child theme can be used as a workaround. It can be cumberstone though as there are many templates referencing `product-list.tpl`.

**Comment by rmilecki** [comment]
It's relevant to me, thanks

**Comment by paulnoelcholot** [comment]
Hello @rmilecki 

Thank you for your feedback, could you please let us know if you are able to reproduce the bug on a maintained version (9.1.x and 9.2.x)?

Thanks!

**Comment by boo-code** [comment]
@paulnoelcholot - yes, still reproduces on 9.2.x (Smarty 4.5.7). Reproduced the mechanism directly rather than through a theme install, using the same two template directories and the same `parent:` resource that `config/smartyfront.config.inc.php` registers:

```
CONTROL - parent theme only
   product-list.tpl  => 'PARENT-PRODUCT-LIST'
   category.tpl      => 'CATEGORY-CONTENT'

CHILD THEME ACTIVE (child adds only product-list.tpl)
   product-list.tpl  => 'CHILD-PREPEND PARENT-PRODUCT-LIST'      <- intended
   category.tpl      => 'CATEGORY-CONTENTPARENT-PRODUCT-LIST'    <- broken
```

So adding one file to a child theme leaks the parent's `product-list` content into every template that extends it, and the child's own text is dropped from those pages. That matches @rmilecki's third screenshot.

Narrowing it further, the trigger is specifically `prepend` / `append` on the intermediate template, not the child theme mechanism itself:

```
child block: prepend        category.tpl => 'CATEGORY-CONTENTPARENT-PRODUCT-LIST'   BROKEN
child block: append         category.tpl => 'PARENT-PRODUCT-LISTCATEGORY-CONTENT'   BROKEN
child block: plain override category.tpl => 'CATEGORY-CONTENT'                      OK
child: unrelated block      category.tpl => 'CATEGORY-CONTENT'                      OK
child: extends only         category.tpl => 'CATEGORY-CONTENT'                      OK
```

What happens is that `category.tpl` extends `catalog/listing/product-list.tpl` by plain name, and with a child theme active that name resolves to the **child's** file, so a two-level chain silently becomes three. Smarty's `prepend`/`append` then merge across that third level and pull the grandparent's block into the grandchild.

There is a drop-in workaround that gives the identical intended result without touching dependent templates:

```smarty
{* breaks category.tpl and anything else extending product-list.tpl *}
{block name='content' prepend}
  <h1>Child Theme test</h1>
{/block}

{* same output on product-list.tpl, category.tpl stays correct *}
{block name='content'}
  <h1>Child Theme test</h1>
  {$smarty.block.parent}
{/block}
```

Measured side by side: `product-list.tpl` renders `'CHILD-TEXT PARENT-PRODUCT-LIST'` either way, while `category.tpl` renders `'CATEGORY-CONTENT'` only with the second form.

Worth noting that the parent/child page in devdocs teaches the `prepend` form, which is the one that breaks, so that page is where a fix would do the most good. I have not opened a core PR because the merging behaviour is Smarty's own, not something core patches - but if maintainers would rather core did something about it (for instance having parent templates extend `parent:` explicitly), say so and I will look into it.
